src/components/autocomplete/js/autocompleteController.js

```diff
@@ -638,14 +639,17 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
     if (angular.isArray(items)) {
       handleResults(items);
     } else if (items) {
+      fetchesInProgress++;
       setLoading(true);
       promiseFetch = true;
       $mdUtil.nextTick(function () {
         if (items.success) items.success(handleResults);
         if (items.then)    items.then(handleResults);
         if (items.finally) items.finally(function () {
-          setLoading(false);
-          promiseFetch = false;
+          if (--fetchesInProgress === 0) {
+            setLoading(false);
+            promiseFetch = false;
```

Comment:
I think it would be cleaner to have only one variable that tracks the fetching state. I'd get rid of promiseFetch and just use fetchesInProgress. If need be, add a getter method that just returns fetchesInProgress === 0